var/spack/repos/builtin/packages/py-cryptography/package.py

```diff
@@ -0,0 +1,46 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+#
+from spack import *
+
+
+class PyCryptography(PythonPackage):
+    """cryptography is a package which provides cryptographic recipes
+       and primitives to Python developers"""
+
+    homepage = "https://pypi.python.org/pypi/cryptography"
+    url      = "https://pypi.io/packages/source/c/cryptography/cryptography-1.8.1.tar.gz"
+
+    version('1.8.1', '9f28a9c141995cd2300d0976b4fac3fb')
+
+    # dependencies taken from https://github.com/pyca/cryptography/blob/master/setup.py
+    depends_on('py-setuptools@20.5:',   type='build')
+    depends_on('py-cffi@1.4.1:',        type=('build', 'run'))
+    depends_on('py-asn1crypto@0.21.0:', type=('build', 'run'))
+    depends_on('py-six@1.4.1:',         type=('build', 'run'))
+    depends_on('py-idna@2.1:',          type=('build', 'run'))
+    depends_on('py-enum34',             type=('build', 'run'), when='^python@:3.4')
+    depends_on('py-ipaddress',          type=('build', 'run'), when='^python@:3.3')
```

Comment:
If I remember correctly, it depends on whether it needs to be present by default (Python 2.7) or not (Python 3). Basically, if you explicitly specify the version, it will always work. But sometimes, if you don't specify the version and let it choose the latest version of Python 2, it won't work. But if `spack spec py-cryptography` works, then it works.